src/client/Microsoft.Identity.Client/Internal/Requests/RequestBase.cs

```diff
@@ -193,6 +198,32 @@ protected async Task<AuthenticationResult> CacheTokenResponseAndCreateAuthentica
                 msalTokenResponse.SpaAuthCode);
         }
 
+        private AuthenticationResult GetAuthenticationResultFromTokenResponse(
+            MsalTokenResponse msalTokenResponse, 
+            MsalIdTokenCacheItem idTokenCacheItem, 
+            IAccount account)
+        {
+            if (account == null)
+            {
+                bool isAdfsAuthority = AuthenticationRequestParameters.AuthorityInfo.AuthorityType == AuthorityType.Adfs;
+                string preferredUsername = TokenResponseHelper.GetPreferredUsernameFromIdToken(isAdfsAuthority, idTokenCacheItem?.IdToken);
+                string username = isAdfsAuthority ? idTokenCacheItem?.IdToken?.Upn : preferredUsername;
+                string homeAccountId = TokenResponseHelper.GetHomeAccountId(AuthenticationRequestParameters, msalTokenResponse, idTokenCacheItem?.IdToken);
+                Dictionary<string, string> wamAccountIds = TokenResponseHelper.GetWamAccountIds(AuthenticationRequestParameters, msalTokenResponse);
+
+                account = new Account(
```

Comment:
by re-creating the account, you lose out on tenant profiles.

Maybe we don't need a new ctor for `AuthenticationResult`. You already have idTokenCacheItem and account object. Maybe you can just create a simple AccessTokenCacheItem object and use the existing AuthenticationResult ctor?

Reply:
I am only creating the account here when the token cache cannot create it and it is null. under normal circumstances, the account will already be created and have the tenant profiles

We have similar logic when using the other constructor https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/9c881e80995f0fcd04cb684181d8e9e99482f0ae/src/client/Microsoft.Identity.Client/AuthenticationResult.cs#L147-L157